src/cons_list.rs

```diff
@@ -39,10 +31,9 @@ impl<T, Ts: ConsListT<T>> ConsListT<T> for Cons<T, Ts> {
     unsafe fn take_unchecked(&mut self, i: usize) -> T {
         debug_assert!(i < Self::LEN, "Index out of bounds");
         let head = self as *mut Self;
-        let head = head.cast::<ManuallyDrop<T>>();
+        let head = head.cast::<T>();
         let elem = head.add(i);
-        let elem = &mut *elem;
-        ManuallyDrop::take(elem)
+        std::ptr::read(elem)
```

Comment:
I guess I felt a bit uneasy about `std::ptr::read`, but actually this is fine

Reply:
miri reports no undefined behavior. (which wasn't true before this change).